modules/sharedaddy/sharedaddy.php

```diff
@@ -22,6 +22,26 @@ function sharing_email_send_post( $data ) {
 	/** This filter is documented in core/src/wp-includes/pluggable.php */
 	$from_email = apply_filters( 'wp_mail_from', 'wordpress@' . $sitename );
 
+	if ( ! empty( $data['name'] ) ) {
+		$s_name = (string) $data['name'];
+		$name_needs_encoding_regex =
+			'/[' .
+				// SpamAssasin's list of characters which "need MIME" encoding
+				'\x00-\x08\x0b\x0c\x0e-\x1f\x7f-\xff' .
+				// Our list of "unsafe" characters
+				'<\r\n' .
+			']/';
+
+		$needs_encoding = preg_match( $name_needs_encoding_regex, $s_name ) ||      // If it contains any blacklisted chars,
+				! function_exists( 'mb_convert_encoding' ) ||           // Or if we can't use `mb_convert_encoding` encode everything,
+				mb_convert_encoding( $data['name'], 'ASCII' ) !== $s_name // Or if it's not already ASCII
+			);
```

Comment:
I'm getting the following fatal:

`2017/03/28 12:09:48 [error] 14282#0: *231580 FastCGI sent in stderr: "PHP message: PHP Parse error:  syntax error, unexpected ')' in /var/www/sometestsite.com/wp-content/plugins/jetpack/modules/sharedaddy/sharedaddy.php on line 38" while reading response header from upstream, client: 73.204.197.29, server: sometestsite.com, request: "GET / HTTP/1.1", upstream: "fastcgi://unix:/var/run/php-fpm/php-fpm.sock:", host: "sometestsite.com"`

 Looks like we don't need the parenthesis here.